framework-generators/messaging-adapter-generator/src/main/java/uk/gov/justice/raml/jms/core/EventSourcesDestinationTypeFinder.java

```diff
@@ -0,0 +1,108 @@
+package uk.gov.justice.raml.jms.core;
+
+import static java.util.Optional.empty;
+
+import uk.gov.justice.services.generators.subscription.parser.EventSourcesFileParser;
+import uk.gov.justice.services.generators.subscription.parser.EventSourcesFileParserFactory;
+import uk.gov.justice.services.generators.subscription.parser.JmsUriToDestinationConverter;
+import uk.gov.justice.subscription.domain.eventsource.EventSourceDefinition;
+import uk.gov.justice.subscription.jms.core.JmsUriToDestinationTypeConverter;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.jms.Destination;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * In the past, when processing raml files for generating MDBs, the envent-sources.yaml files where jms_uri are defined were ignored.
+ * This class is allowing to read any event-sources.yaml files in the classpath and trying to find the corresponding jms_uri
+ * that is used to determine whether we are dealing with a topic or a queue.
+ * Note that this is only used for event-processors only
+ */
+public class EventSourcesDestinationTypeFinder {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventSourcesDestinationTypeFinder.class);
+
+    private final JmsUriToDestinationTypeConverter uriToDestinationTypeConverter = new JmsUriToDestinationTypeConverter();
+    JmsUriToDestinationConverter jmsUriToDestinationConverter = new JmsUriToDestinationConverter();
+    private final EventSourcesFileParserFactory eventSourceYamlParserFactory = new EventSourcesFileParserFactory();
+
+
+    /**
+     * @param serviceComponent EVENT_PROCESSOR, etc
+     * @param destinationName  the matching destination name
+     * @return Queue, Topic or empty
+     */
+    public Optional<Class<? extends Destination>> findForEventProcessor(final String serviceComponent, final String destinationName) {
+
+        if (serviceComponent == null || destinationName == null || !uriToDestinationTypeConverter.isEventProcessor(serviceComponent)) {
+            return empty();
+        }
+
+        try {
+            final EventSourcesFileParser eventSourcesFileParser = eventSourceYamlParserFactory.create();
+            final ArrayList<Path> list = new ArrayList<>();
+            final List<EventSourceDefinition> eventSourceDefinitionList = eventSourcesFileParser.getEventSourceDefinitions(Path.of(""), list);
+            if (eventSourceDefinitionList == null) {
+                LOGGER.debug("No SubscriptionWrapper found .. returning");
```

Comment:
Do we need these debug log messages? I'm guessing you added them to make sure stuff is working, but now that it is, are they still needed? If they are then the logger should be converted to a member variable (i.e. not static), mocked in the tests and verified

Reply:
I have removed those debug statements